src/antlr/LineageUtil.ts

```diff
@@ -0,0 +1,54 @@
+import { CharStreams, CommonTokenStream } from "antlr4ts";
+import { Lineage } from "../Lineage";
+import { QueryVisitor } from "./QueryVisitor";
+import { SqlBaseLexer } from "./SqlBaseLexer";
+import { SqlBaseParser } from "./SqlBaseParser";
+import { UppercaseCharStream } from "./UppercaseCharStream";
+import { LineageContext } from "./LineageContext";
+
+export function getLineage<TableData extends { id: string }, ColumnData extends { id: string }>(
+  sql: string,
+  getters: {
+    getTable: (tableId: string) => TableData;
+    getColumns: (tableId: string) => ColumnData[];
+  }
```

Comment:
Before I forget, we can simplify this to have only one getter.

```suggestion
getTable: (tableId: string) => TableData & { columns: ColumnData[] }
```

And, because this interface will not work for multiple catalogs, we need to extend this getter:

```ts
type TablePrimary {
  catalogName?: string;
  schemaName?: string;
  tableName: string;
}
```
```suggestion
getTable: (table: TablePrimary) => TableData & { columns: ColumnData[] }
```

(I'm still playing on the integration side, but this one of the point I missed on my previous iteration 😉)

Reply:
Thinking how to extract table data from the returned metadata to put into `Table` lineage? Probably need to return `{tableData: TableData, columns: ColumnData[]}`.

Reply:
We can do `omit(data, "columns")` should be good enough (with `import omit from "lodash/omit"`)